bundles/automation/org.eclipse.smarthome.automation.providers/src/main/java/org/eclipse/smarthome/automation/core/provider/RuleResourceBundleImporter.java

```diff
@@ -0,0 +1,160 @@
+/*******************************************************************************
+ * Copyright (c) 1997, 2015 by ProSyst Software GmbH
+ * http://www.prosyst.com
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *    ProSyst Software GmbH - initial API and implementation
+ *******************************************************************************/
+
+package org.eclipse.smarthome.automation.core.provider;
+
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.net.URL;
+import java.util.Enumeration;
+import java.util.Iterator;
+import java.util.Set;
+
+import org.eclipse.smarthome.automation.Rule;
+import org.eclipse.smarthome.automation.RuleRegistry;
+import org.eclipse.smarthome.automation.handler.parser.Parser;
+import org.eclipse.smarthome.automation.handler.parser.Status;
+import org.osgi.framework.Bundle;
+import org.osgi.framework.BundleContext;
+import org.osgi.framework.ServiceReference;
+import org.osgi.util.tracker.ServiceTracker;
+
+/**
+ * @author Ana Dimova - Initial Contribution
+ *
+ */
+public class RuleResourceBundleImporter extends GeneralResourceBundleProvider {
+
+    private RuleRegistry ruleRegistry;
+    private ServiceTracker adminTracker;
+
+    /**
+     * @param bc
+     */
+    public RuleResourceBundleImporter(BundleContext context) {
+        super(context);
+        path = PATH + "/rules/";
+        adminTracker = new ServiceTracker(context, RuleRegistry.class.getName(), this);
+        adminTracker.open();
+    }
+
+    @Override
+    public void close() {
+        if (adminTracker != null) {
+            adminTracker.close();
+            adminTracker = null;
+        }
+        super.close();
+    }
+
+    /**
+     * @see org.eclipse.smarthome.automation.core.provider.GeneralResourceBundleProvider#addingService(ServiceReference)
+     */
+    @Override
+    public Object addingService(ServiceReference reference) {
+        Object service = bc.getService(reference);
+        if (service instanceof RuleRegistry) {
+            ruleRegistry = (RuleRegistry) service;
+            return service;
```

Comment:
Hm, don't you have to check for waitingProviders here? If all system is started up and the ruleRegistry service comes in last, it would not result in any processing otherwise.